app/src/main/java/com/simpragma/recipe/recipeapplication1/SearchItemActivity.java

```diff
@@ -0,0 +1,269 @@
+
+    EditText eidiText;
+    Button button;
+    String TAG=getClass().getSimpleName();
+    ArrayList<Result> results;
+    private ArrayList<Result> mFilteredList;
+    Recipie post;
+    RecyclerView recyclerView;
+    private DataAdapter customAdapter;
+    DataBaseAdapter dataBaseAdapter;
+    ArrayList<DataBaseResult> dataBaseResults;
+    DatabaseHandler db;
+    DataBaseResult dataBaseResult;
+    String text,url;
+    TextView textView;
+
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_search_item1);
+
+        if(Constants1.Type.Staging == Constants1.Type.Staging)
+
+        {
+            Log.d(TAG,"FREE VERSION");
+        }
+        else
+        {
+            Log.d(TAG," PAID VERSION");
+        }
+
+
+        eidiText=(EditText) findViewById(R.id.editText_search);
+        button=(Button) findViewById(R.id.button_search);
+        textView=(TextView) findViewById(R.id.textnotfound);
+        recyclerView = (RecyclerView) findViewById(R.id.card_recycler_view);
+        recyclerView.setLayoutManager(new GridLayoutManager(SearchItemActivity.this,2));
+        db = new DatabaseHandler(this);
+
+
+        button.setOnClickListener(this);
+        //requestJsonObject();
+
+//        Intent intent = getIntent();
+//        startActivity(intent);
+        dataBaseResults = db.getAllContacts();
+        Log.d("DataBase Result",dataBaseResults.size()+"");
+        if(dataBaseResults.size()>0)
+        {
+            dataBaseAdapter=new DataBaseAdapter(SearchItemActivity.this,dataBaseResults);
+            recyclerView.setBackgroundColor(Color.LTGRAY);
+            recyclerView.setAdapter(dataBaseAdapter);
+        }
+
+    }
+
+    @Override
+    protected void onStart() {
+        super.onStart();
+
+         //requestJsonObject();
+       // button.setOnClickListener(this);
+
+        dataBaseResults = db.getAllContacts();
+        Log.d("DataBase Result",dataBaseResults.size()+"");
+        if(dataBaseResults.size()>0)
+        {
+            dataBaseAdapter=new DataBaseAdapter(SearchItemActivity.this,dataBaseResults);
+            recyclerView.setAdapter(dataBaseAdapter);
+        }
+
+        for (DataBaseResult cn : dataBaseResults) {
+            String log = "Id: "+cn.getId()+" ,Title: " + cn.getTitle()
+                    + " ,Ingredients: " + cn.getIngredients() +"Href"+ cn.getHref()+ " ,Thumbnail"+ cn.getThumbnail();
+            // Writing Contacts to log
+            Log.d("Name: ", log);
+        }
+
+    }
+
+    @Override
+    protected void onRestart() {
+        super.onRestart();
+        Intent intent = getIntent();
+        startActivity(intent);
+    }
+
+    @Override
+    protected void onResume() {
+        super.onResume();
+
+    }
+
+    private void requestJsonObject(){
+
+        text=eidiText.getText().toString();
+
+        Uri uri = Uri.parse("http://www.recipepuppy.com/api/?i="+text+"");
```

Comment:
Please move API url to configuration file.

Reply:
Don't hardcode it here.

Reply:
i put url in configuration file